### DISCUSSION TITLE
GraphQL mutation with relations

### DISCUSSION THREAD

**Discussion by clck-dr** [directus/directus#7185] [Feature Requests]
I had some problems with mutations on create item with relations.

I want to create an item with 1 to n relation by invoking following request:

POST /graphql

```
mutation {
    create_activity_item(
        data: {
            start: "2021-08-04T07:06:10Z"
            time: "00:01:11"
            calculatedpoints: 0
            activitytype: "1" 
        },
    )     
}
```

The field activitytype is a different collection. The activitytypes ID is available.
I get an error that I have to request activitytype as create_activitytype_input. I don't want to create a new activitytype but use a allready existing on by setting ID.

Is there a solution I don't see?

**Comment by rijkvanzanten** [comment]
@clck-dr GraphQL doesn't allow for two different types on the same field. In this case, `activitytype` can both be a nested object _or_ a primary key in Directus, but can only be a nested object in GraphQL. To select the existing item, simply use the existing primary key as part of that object in the payload:

```graphql
mutation {
    create_activity_item(
        data: {
            start: "2021-08-04T07:06:10Z"
            time: "00:01:11"
            calculatedpoints: 0
            activitytype: {
              id: 1
            } 
        },
    )     
}
```

(i'm assuming your primary key is called `id`, make sure to update the above example if not)

**Reply by clck-dr** [reply]
Thank you very much for this solution.

If I do it that way, I have to add the field "name" as required field and status too.
So I'd figured out two options:
1) Define no required fields to collections.
2) Always add all the activitytype fields to the request.

What do you suggest as preferable solution?

**Reply by rijkvanzanten** [reply]
Oh good point. That activitytype record should be checked against the _update_ permissions, not the _create_ permissions in this case. Mind opening a new issue?

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Comment by dp1194** [comment]
Admittedly we don't do a lot of mutations through the GraphQL API but now that we have a project where we'd like to use GQL mutations we've hit this hard. I'm surprised more people haven't raised this issue.

Is it still the case that the checks are happening against the create permissions?

I'd be happy to open a new issue but is it a feature request or a bug?
